### PR TITLE
Add Redux Toolkit and Configure Redux Store

### PR DESCRIPTION
**Pull request by kessie2862**
## Description
This PR adds `Redux` Toolkit to the Bookstore project, sets up the `Redux` directory structure, and configures the `Redux` store. Additionally, it defines two slices of state: one for books and another for categories. The book slice includes `reducers` to add and remove books, while the categories slice includes a `reducer` that always returns the string "Under construction" as the initial state.

## Changes

- [ ] Added `Redux` Toolkit to the project 
- [ ] Created a new directory to contain all `Redux` logic
- [ ] Configured the `Redux` store
- [ ] Defined a slice of state for books that includes an array of books, and `reducer` that adds and removes a book.
- [ ] Defined a slice of state for categories that includes an array of categories and a `reducer` that checks the status and always returns "Under construction"

### CODE REVIEW

File: src/redux/bookSlice.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const bookSlice = createSlice({
+  name: 'books',
+  initialState: [],
+  reducers: {
+    addBook: (state, action) => {
+      state.push(action.payload);
```
Comment (Ismailco) [inline src/redux/bookSlice.js]: - [ ] [OPTIONAL] Kindly make sure the state is immutable you can use `{...state.books, action.payload}` instead

File: src/redux/categorySlice.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const categorySlice = createSlice({
+  name: 'categories',
+  initialState: 'Under construction',
```
Comment (Ismailco) [inline src/redux/categorySlice.js]: - [ ] The `initialState` value should be `categories` that contain an empty array, kindly make sure to implement this like so 👇🏼 
```
 initialState: {
    categories: [],
  },
```

File: src/redux/categorySlice.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+  reducers: {},
```
Comment (Ismailco) [inline src/redux/categorySlice.js]: - [ ] Kindly make sure to create one reducer `checkStatus` and export it as an action

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @kessie2862, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
- [ ] The project should follow the `ducks pattern` to structure files, kindly consider adding a folder for every slice. it should look like this 👉🏼  `redux/books/bookSlice.js` 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categorySlice.js (hunk 3 above)
Comment (kessie2862) [reply src/redux/categorySlice.js]: Thanks. I've modified the code to include the `checkStatus` reducer

File: src/redux/categorySlice.js (hunk 2 above)
Comment (kessie2862) [reply src/redux/categorySlice.js]: I just implemented that. Thanks

File: src/redux/bookSlice.js (hunk 1 above)
Comment (kessie2862) [reply src/redux/bookSlice.js]: I've implemented the suggestion. Thanks

**Review by prolajumokeoni — APPROVED** [review]
Hi @kessie2862 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
 - All changes requested have been implemented 🥇  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**